### PR TITLE
Feature | PEO-1017

### PR DESCRIPTION
**Pull request by Bulatkin**
JIRA ticket - https://garage.atlassian.net/browse/PEO-1017

### Before submitting the merge request make sure the following are checked
- [ ] Followed the style guidelines of this project
- [ ] Performed a self-review of own code
- [ ] Wrote the tests that prove that fix is effective/that feature works
- [ ] Check correctness of commit names/branch name/assignees list/labels
### Notable changes
- Added CircleCI configuration
- Fixed gem version number
### Commands to run
- If PR contains Rake tasks/seeds/credentials changes - describe them below with a list

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            bundle install --jobs=4 --retry=3 --path vendor/bundle
+
+      - run:
+          name: Rubocop
+          command: bundle exec rubocop
+
+      - run:
+          name: Fasterer
+          command: bundle exec fasterer
+
```
Comment (Volosh1n) [inline .circleci/config.yml:45]: I'd suggest to move it to another [workflow](https://github.com/rubygarage/circledge/blob/master/intro_to_circleci.md#workflows) 'linters'.

**Review by Volosh1n — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+          path: tmp/rspec
+      - store_artifacts:
+          path: /tmp/test-results
+          destination: test-results
+workflows:
+  version: 2
+  build_and_test:
+    jobs:
+      - build
+      - test
```
Comment (Volosh1n) [inline .circleci/config.yml]:
```yml
workflows:
  version: 2
  build:
    jobs:
      - lintering
      - specs
        requires:
          - lintering
```
It should be two separated jobs - one for lintering, another one for specs.
One job requires another - second job shouldn't start if first job failed.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          command: |
+            mkdir -p /tmp/test-results
+            circleci tests glob "spec/**/*_spec.rb" > test_files.txt
+            cat test_files.txt | circleci tests split --split-by=timings > split_test_files.txt
+
+            TESTFILES="$(cat split_test_files.txt | xargs)"
+            if [ -z "$TESTFILES" ]; then echo  'No tests are scheduled to be executed on this node'; exit 0; fi
+
+            export SPEC="$TESTFILES"
+            export SPEC_OPTS="--format documentation --format RspecJunitFormatter --out tmp/rspec/rspec-${CIRCLE_NODE_INDEX:-0}.xml"
```
Comment (Volosh1n) [inline .circleci/config.yml]: what is it? Commands should be small

**Review by Volosh1n — CHANGES_REQUESTED** [review]

**Comment by Bulatkin** [comment]
Moved changes to Feature | PEO-1023

**Pull request closed without merging**